<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Bioconductor_GEO_Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="GEO_Project_code_files/libs/clipboard/clipboard.min.js"></script>
<script src="GEO_Project_code_files/libs/quarto-html/quarto.js"></script>
<script src="GEO_Project_code_files/libs/quarto-html/popper.min.js"></script>
<script src="GEO_Project_code_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="GEO_Project_code_files/libs/quarto-html/anchor.min.js"></script>
<link href="GEO_Project_code_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="GEO_Project_code_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="GEO_Project_code_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="GEO_Project_code_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="GEO_Project_code_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Bioconductor_GEO_Project</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># The following initializes most up to date version of Bioc</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"NanoStringNCTools"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Bioconductor version 3.16 (BiocManager 1.30.19), R 4.2.2 (2022-10-31)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'NanoStringNCTools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Old packages: 'ggiraph'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"GeomxTools"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Bioconductor version 3.16 (BiocManager 1.30.19), R 4.2.2 (2022-10-31)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'GeomxTools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Old packages: 'ggiraph'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"GeoMxWorkflows"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Bioconductor version 3.16 (BiocManager 1.30.19), R 4.2.2 (2022-10-31)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'GeoMxWorkflows'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Old packages: 'ggiraph'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NanoStringNCTools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'NanoStringNCTools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocManager':

    version</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeomxTools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeoMxWorkflows)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">packageVersion</span>(<span class="st">"GeomxTools"</span>) <span class="sc">&lt;</span> <span class="st">"2.1"</span> <span class="sc">&amp;</span> </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>   <span class="fu">packageVersion</span>(<span class="st">"GeoMxWorkflows"</span>) <span class="sc">&gt;=</span> <span class="st">"1.0.1"</span>){</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"GeomxTools and Workflow versions do not match. Please use the same version. </span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="st">    This workflow is meant to be used with most current version of packages. </span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="st">    If you are using an older version of Bioconductor please reinstall GeoMxWorkflows and use vignette(GeoMxWorkflows) instead"</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">packageVersion</span>(<span class="st">"GeomxTools"</span>) <span class="sc">&gt;</span> <span class="st">"2.1"</span> <span class="sc">&amp;</span> </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>   <span class="fu">packageVersion</span>(<span class="st">"GeoMxWorkflows"</span>) <span class="sc">&lt;=</span> <span class="st">"1.0.1"</span>){</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"GeomxTools and Workflow versions do not match. </span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="st">         Please use the same version, see install instructions above."</span>)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># to remove current package version</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove.packages("GeomxTools")</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove.packages("GeoMxWorkflows")</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># see install instructions above </span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>datadir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"WTA_NGS_Example"</span>,</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">package=</span><span class="st">"GeoMxWorkflows"</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>DCCFiles <span class="ot">&lt;-</span> <span class="fu">dir</span>(<span class="fu">file.path</span>(datadir, <span class="st">"dccs"</span>), <span class="at">pattern =</span> <span class="st">".dcc$"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>SampleAnnotationFile <span class="ot">&lt;-</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir</span>(<span class="fu">file.path</span>(datadir, <span class="st">"annotation"</span>), <span class="at">pattern =</span> <span class="st">".xlsx$"</span>,</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>PKCFiles <span class="ot">&lt;-</span> <span class="fu">unzip</span>(<span class="at">zipfile =</span> <span class="fu">dir</span>(<span class="fu">file.path</span>(datadir, <span class="st">"pkcs"</span>), <span class="at">pattern =</span> <span class="st">".zip$"</span>,</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>                                <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">readNanoStringGeoMxSet</span>(<span class="at">dccFiles =</span> DCCFiles,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">pkcFiles =</span> PKCFiles,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">phenoDataFile =</span> SampleAnnotationFile,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">phenoDataSheet =</span> <span class="st">"Template"</span>,</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">phenoDataDccColName =</span> <span class="st">"Sample_ID"</span>,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">protocolDataColNames =</span> <span class="fu">c</span>(<span class="st">"aoi"</span>, <span class="st">"roi"</span>),</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">experimentDataColNames =</span> <span class="fu">c</span>(<span class="st">"panel"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>pkcs <span class="ot">&lt;-</span> <span class="fu">annotation</span>(demoData)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".pkc"</span>, <span class="st">""</span>, pkcs)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">data.frame</span>(<span class="at">PKCs =</span> pkcs, <span class="at">modules =</span> modules))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">PKCs</th>
<th style="text-align: left;">modules</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Hsa_WTA_v1.0.pkc</td>
<td style="text-align: left;">Hsa_WTA_v1.0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NanoStringNCTools':

    groups</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    first, intersect, rename, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co"># select the annotations we want to show, use `` to surround column names with</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># spaces or special symbols</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">count</span>(<span class="fu">pData</span>(demoData), <span class="st">`</span><span class="at">slide name</span><span class="st">`</span>, class, region, segment)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co"># simplify the slide names</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>count_mat<span class="sc">$</span><span class="st">`</span><span class="at">slide name</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"disease"</span>, <span class="st">"d"</span>,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">gsub</span>(<span class="st">"normal"</span>, <span class="st">"n"</span>, count_mat<span class="sc">$</span><span class="st">`</span><span class="at">slide name</span><span class="st">`</span>))</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co"># gather the data and plot in order: class, slide name, region, segment</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>test_gr <span class="ot">&lt;-</span> <span class="fu">gather_set_data</span>(count_mat, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>test_gr<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">factor</span>(test_gr<span class="sc">$</span>x,</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"class"</span>, <span class="st">"slide name"</span>, <span class="st">"region"</span>, <span class="st">"segment"</span>))</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Sankey</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(test_gr, <span class="fu">aes</span>(x, <span class="at">id =</span> id, <span class="at">split =</span> y, <span class="at">value =</span> n)) <span class="sc">+</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets</span>(<span class="fu">aes</span>(<span class="at">fill =</span> region), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">axis.width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets_axes</span>(<span class="at">axis.width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets_labels</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">17</span>) <span class="sc">+</span> </span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"segment"</span>, <span class="at">x =</span> <span class="fl">4.25</span>, <span class="at">xend =</span> <span class="fl">4.25</span>,</span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> <span class="dv">20</span>, <span class="at">yend =</span> <span class="dv">120</span>, <span class="at">lwd =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">4.19</span>, <span class="at">y =</span> <span class="dv">70</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">5</span>,</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">label =</span> <span class="st">"100 segments"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets_axes()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets_axes()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shift counts to one</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span> <span class="fu">shiftCountsOne</span>(demoData, <span class="at">useDALogic =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>QC_params <span class="ot">&lt;-</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">minSegmentReads =</span> <span class="dv">1000</span>, <span class="co"># Minimum number of reads (1000)</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">percentTrimmed =</span> <span class="dv">80</span>,    <span class="co"># Minimum % of reads trimmed (80%)</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">percentStitched =</span> <span class="dv">80</span>,   <span class="co"># Minimum % of reads stitched (80%)</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">percentAligned =</span> <span class="dv">75</span>,    <span class="co"># Minimum % of reads aligned (80%)</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">percentSaturation =</span> <span class="dv">50</span>, <span class="co"># Minimum sequencing saturation (50%)</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">minNegativeCount =</span> <span class="dv">1</span>,   <span class="co"># Minimum negative control counts (10)</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">maxNTCCount =</span> <span class="dv">9000</span>,     <span class="co"># Maximum counts observed in NTC well (1000)</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">minNuclei =</span> <span class="dv">20</span>,         <span class="co"># Minimum # of nuclei estimated (100)</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">minArea =</span> <span class="dv">1000</span>)         <span class="co"># Minimum segment area (5000)</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setSegmentQCFlags</span>(demoData, </span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">qcCutoffs =</span> QC_params)        </span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate QC Results</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>QCResults <span class="ot">&lt;-</span> <span class="fu">protocolData</span>(demoData)[[<span class="st">"QCFlags"</span>]]</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>flag_columns <span class="ot">&lt;-</span> <span class="fu">colnames</span>(QCResults)</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>QC_Summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Pass =</span> <span class="fu">colSums</span>(<span class="sc">!</span>QCResults[, flag_columns]),</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Warning =</span> <span class="fu">colSums</span>(QCResults[, flag_columns]))</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>QCResults<span class="sc">$</span>QCStatus <span class="ot">&lt;-</span> <span class="fu">apply</span>(QCResults, 1L, <span class="cf">function</span>(x) {</span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(<span class="fu">sum</span>(x) <span class="sc">==</span> 0L, <span class="st">"PASS"</span>, <span class="st">"WARNING"</span>)</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>QC_Summary[<span class="st">"TOTAL FLAGS"</span>, ] <span class="ot">&lt;-</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="fu">sum</span>(QCResults[, <span class="st">"QCStatus"</span>] <span class="sc">==</span> <span class="st">"PASS"</span>),</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">sum</span>(QCResults[, <span class="st">"QCStatus"</span>] <span class="sc">==</span> <span class="st">"WARNING"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>col_by <span class="ot">&lt;-</span> <span class="st">"segment"</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Graphical summaries of QC statistics plot function</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>QC_histogram <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">assay_data =</span> <span class="cn">NULL</span>,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">annotation =</span> <span class="cn">NULL</span>,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">fill_by =</span> <span class="cn">NULL</span>,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">thr =</span> <span class="cn">NULL</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">scale_trans =</span> <span class="cn">NULL</span>) {</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>    plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(assay_data,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"unlist(`"</span>, annotation, <span class="st">"`)"</span>),</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>                             <span class="at">fill =</span> fill_by)) <span class="sc">+</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> thr, <span class="at">lty =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">facet_wrap</span>(<span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"~"</span>, fill_by)), <span class="at">nrow =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> annotation, <span class="at">y =</span> <span class="st">"Segments, #"</span>, <span class="at">title =</span> annotation)</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(scale_trans)) {</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>        plt <span class="ot">&lt;-</span> plt <span class="sc">+</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> scale_trans)</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    plt</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"Trimmed (%)"</span>, col_by, <span class="dv">80</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation ideoms with `aes()`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"Stitched (%)"</span>, col_by, <span class="dv">80</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"Aligned (%)"</span>, col_by, <span class="dv">75</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"Saturated (%)"</span>, col_by, <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Sequencing Saturation (%)"</span>,</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Sequencing Saturation (%)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"area"</span>, col_by, <span class="dv">1000</span>, <span class="at">scale_trans =</span> <span class="st">"log10"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">QC_histogram</span>(<span class="fu">sData</span>(demoData), <span class="st">"nuclei"</span>, col_by, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the negative geometric means for each module</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>negativeGeoMeans <span class="ot">&lt;-</span> </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">esBy</span>(<span class="fu">negativeControlSubset</span>(demoData), </span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">GROUP =</span> <span class="st">"Module"</span>, </span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">FUN =</span> <span class="cf">function</span>(x) { </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">assayDataApply</span>(x, <span class="at">MARGIN =</span> <span class="dv">2</span>, <span class="at">FUN =</span> ngeoMean, <span class="at">elt =</span> <span class="st">"exprs"</span>) </span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>         }) </span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="fu">protocolData</span>(demoData)[[<span class="st">"NegGeoMean"</span>]] <span class="ot">&lt;-</span> negativeGeoMeans</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a><span class="co"># explicitly copy the Negative geoMeans from sData to pData</span></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>negCols <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"NegGeoMean_"</span>, modules)</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(demoData)[, negCols] <span class="ot">&lt;-</span> <span class="fu">sData</span>(demoData)[[<span class="st">"NegGeoMean"</span>]]</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ann <span class="cf">in</span> negCols) {</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>    plt <span class="ot">&lt;-</span> <span class="fu">QC_histogram</span>(<span class="fu">pData</span>(demoData), ann, col_by, <span class="dv">2</span>, <span class="at">scale_trans =</span> <span class="st">"log10"</span>)</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(plt)</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># detatch neg_geomean columns ahead of aggregateCounts call</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(demoData) <span class="ot">&lt;-</span> <span class="fu">pData</span>(demoData)[, <span class="sc">!</span><span class="fu">colnames</span>(<span class="fu">pData</span>(demoData)) <span class="sc">%in%</span> negCols]</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="co"># show all NTC values, Freq = # of Segments with a given NTC count:</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">table</span>(<span class="at">NTC_Count =</span> <span class="fu">sData</span>(demoData)<span class="sc">$</span>NTC),</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"NTC Count"</span>, <span class="st">"# of Segments"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">NTC Count</th>
<th style="text-align: right;"># of Segments</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">36</td>
</tr>
<tr class="even">
<td style="text-align: left;">113</td>
<td style="text-align: right;">71</td>
</tr>
<tr class="odd">
<td style="text-align: left;">397</td>
<td style="text-align: right;">34</td>
</tr>
<tr class="even">
<td style="text-align: left;">8704</td>
<td style="text-align: right;">94</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(QC_Summary, <span class="at">caption =</span> <span class="st">"QC Summary Table for each Segment"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>QC Summary Table for each Segment</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Pass</th>
<th style="text-align: right;">Warning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LowReads</td>
<td style="text-align: right;">231</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">LowTrimmed</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LowStitched</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">LowAligned</td>
<td style="text-align: right;">229</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LowSaturation</td>
<td style="text-align: right;">231</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">LowNegatives</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HighNTC</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">LowNuclei</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LowArea</td>
<td style="text-align: right;">235</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">TOTAL FLAGS</td>
<td style="text-align: right;">229</td>
<td style="text-align: right;">6</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span> demoData[, QCResults<span class="sc">$</span>QCStatus <span class="sc">==</span> <span class="st">"PASS"</span>]</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Subsetting our dataset has removed samples which did not pass QC</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(demoData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Features  Samples 
   18642      229 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generally keep the qcCutoffs parameters unchanged. Set removeLocalOutliers to </span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="co"># FALSE if you do not want to remove local outliers</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span> <span class="fu">setBioProbeQCFlags</span>(demoData, </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">qcCutoffs =</span> <span class="fu">list</span>(<span class="at">minProbeRatio =</span> <span class="fl">0.1</span>,</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">percentFailGrubbs =</span> <span class="dv">20</span>), </span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">removeLocalOutliers =</span> <span class="cn">TRUE</span>)</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>ProbeQCResults <span class="ot">&lt;-</span> <span class="fu">fData</span>(demoData)[[<span class="st">"QCFlags"</span>]]</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define QC table for Probe QC</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>qc_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Passed =</span> <span class="fu">sum</span>(<span class="fu">rowSums</span>(ProbeQCResults[, <span class="sc">-</span><span class="dv">1</span>]) <span class="sc">==</span> <span class="dv">0</span>),</span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Global =</span> <span class="fu">sum</span>(ProbeQCResults<span class="sc">$</span>GlobalGrubbsOutlier),</span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Local =</span> <span class="fu">sum</span>(<span class="fu">rowSums</span>(ProbeQCResults[, <span class="sc">-</span><span class="dv">2</span><span class="sc">:-</span><span class="dv">1</span>]) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>                                <span class="sc">&amp;</span> <span class="sc">!</span>ProbeQCResults<span class="sc">$</span>GlobalGrubbsOutlier))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Subset object to exclude all that did not pass Ratio &amp; Global testing</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>ProbeQCPassed <span class="ot">&lt;-</span> </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(demoData, </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>           <span class="fu">fData</span>(demoData)[[<span class="st">"QCFlags"</span>]][,<span class="fu">c</span>(<span class="st">"LowProbeRatio"</span>)] <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>               <span class="fu">fData</span>(demoData)[[<span class="st">"QCFlags"</span>]][,<span class="fu">c</span>(<span class="st">"GlobalGrubbsOutlier"</span>)] <span class="sc">==</span> <span class="cn">FALSE</span>)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ProbeQCPassed)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Features  Samples 
   18641      229 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Features  Samples </span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    18641      229</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>demoData <span class="ot">&lt;-</span> ProbeQCPassed </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how many unique targets the object has</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">featureData</span>(demoData)[[<span class="st">"TargetName"</span>]]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18504</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>target_demoData <span class="ot">&lt;-</span> <span class="fu">aggregateCounts</span>(demoData)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(target_demoData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Features  Samples 
   18504      229 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">exprs</span>(target_demoData)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      DSP-1001250007851-H-A02.dcc DSP-1001250007851-H-A03.dcc
A2M                           485                         262
NAT2                           15                          18
ACADM                          31                          15
ACADS                          27                          17
ACAT1                          29                          24</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define LOQ SD threshold and minimum value</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>cutoff <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>minLOQ <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate LOQ per module tested</span></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>LOQ <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names =</span> <span class="fu">colnames</span>(target_demoData))</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(module <span class="cf">in</span> modules) {</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>    vars <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">c</span>(<span class="st">"NegGeoMean_"</span>, <span class="st">"NegGeoSD_"</span>),</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>                   module)</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">all</span>(vars[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">%in%</span> <span class="fu">colnames</span>(<span class="fu">pData</span>(target_demoData)))) {</span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>        LOQ[, module] <span class="ot">&lt;-</span></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>            <span class="fu">pmax</span>(minLOQ,</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">pData</span>(target_demoData)[, vars[<span class="dv">1</span>]] <span class="sc">*</span> </span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">pData</span>(target_demoData)[, vars[<span class="dv">2</span>]] <span class="sc">^</span> cutoff)</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>LOQ <span class="ot">&lt;-</span> LOQ</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>LOQ_Mat <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(module <span class="cf">in</span> modules) {</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>    ind <span class="ot">&lt;-</span> <span class="fu">fData</span>(target_demoData)<span class="sc">$</span>Module <span class="sc">==</span> module</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>    Mat_i <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">esApply</span>(target_demoData[ind, ], <span class="at">MARGIN =</span> <span class="dv">1</span>,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">FUN =</span> <span class="cf">function</span>(x) {</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>                           x <span class="sc">&gt;</span> LOQ[, module]</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>                       }))</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>    LOQ_Mat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(LOQ_Mat, Mat_i)</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ensure ordering since this is stored outside of the geomxSet</span></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>LOQ_Mat <span class="ot">&lt;-</span> LOQ_Mat[<span class="fu">fData</span>(target_demoData)<span class="sc">$</span>TargetName, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save detection rate information to pheno data</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>GenesDetected <span class="ot">&lt;-</span> </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colSums</span>(LOQ_Mat, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>GeneDetectionRate <span class="ot">&lt;-</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>GenesDetected <span class="sc">/</span> <span class="fu">nrow</span>(target_demoData)</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine detection thresholds: 1%, 5%, 10%, 15%, &gt;15%</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>DetectionThreshold <span class="ot">&lt;-</span> </span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cut</span>(<span class="fu">pData</span>(target_demoData)<span class="sc">$</span>GeneDetectionRate,</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="fl">0.15</span>, <span class="dv">1</span>),</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"&lt;1%"</span>, <span class="st">"1-5%"</span>, <span class="st">"5-10%"</span>, <span class="st">"10-15%"</span>, <span class="st">"&gt;15%"</span>))</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a><span class="co"># stacked bar plot of different cut points (1%, 5%, 10%, 15%)</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">pData</span>(target_demoData),</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> DetectionThreshold)) <span class="sc">+</span></span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> region)) <span class="sc">+</span></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">"count"</span>, <span class="fu">aes</span>(<span class="at">label =</span> ..count..), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Detection Rate"</span>,</span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Segments, #"</span>,</span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="st">"Segment Type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The dot-dot notation (`..count..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(count)` instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cut percent genes detected at 1, 5, 10, 15</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">table</span>(<span class="fu">pData</span>(target_demoData)<span class="sc">$</span>DetectionThreshold,</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>class))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">DKD</th>
<th style="text-align: right;">normal</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">&lt;1%</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">1-5%</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5-10%</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">10-15%</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">&gt;15%</td>
<td style="text-align: right;">102</td>
<td style="text-align: right;">94</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>target_demoData <span class="ot">&lt;-</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>    target_demoData[, <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>GeneDetectionRate <span class="sc">&gt;=</span> .<span class="dv">1</span>]</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(target_demoData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Features  Samples 
   18504      221 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select the annotations we want to show, use `` to surround column names with</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="co"># spaces or special symbols</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">count</span>(<span class="fu">pData</span>(demoData), <span class="st">`</span><span class="at">slide name</span><span class="st">`</span>, class, region, segment)</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="co"># simplify the slide names</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>count_mat<span class="sc">$</span><span class="st">`</span><span class="at">slide name</span><span class="st">`</span> <span class="ot">&lt;-</span> </span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">"disease"</span>, <span class="st">"d"</span>,</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">gsub</span>(<span class="st">"normal"</span>, <span class="st">"n"</span>, count_mat<span class="sc">$</span><span class="st">`</span><span class="at">slide name</span><span class="st">`</span>))</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a><span class="co"># gather the data and plot in order: class, slide name, region, segment</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>test_gr <span class="ot">&lt;-</span> <span class="fu">gather_set_data</span>(count_mat, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>test_gr<span class="sc">$</span>x <span class="ot">&lt;-</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(test_gr<span class="sc">$</span>x,</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"class"</span>, <span class="st">"slide name"</span>, <span class="st">"region"</span>, <span class="st">"segment"</span>))</span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Sankey</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(test_gr, <span class="fu">aes</span>(x, <span class="at">id =</span> id, <span class="at">split =</span> y, <span class="at">value =</span> n)) <span class="sc">+</span></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets</span>(<span class="fu">aes</span>(<span class="at">fill =</span> region), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">axis.width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets_axes</span>(<span class="at">axis.width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_parallel_sets_labels</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">17</span>) <span class="sc">+</span> </span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"segment"</span>, <span class="at">x =</span> <span class="fl">4.25</span>, <span class="at">xend =</span> <span class="fl">4.25</span>, <span class="at">y =</span> <span class="dv">20</span>, </span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">yend =</span> <span class="dv">120</span>, <span class="at">lwd =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb78-28"><a href="#cb78-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">4.19</span>, <span class="at">y =</span> <span class="dv">70</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">5</span>,</span>
<span id="cb78-29"><a href="#cb78-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">label =</span> <span class="st">"100 segments"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets_axes()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Computation failed in `stat_parallel_sets_axes()`
Caused by error in `FUN()`:
! id must be unique within axes</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales) <span class="co"># for percent</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate detection rate:</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>LOQ_Mat <span class="ot">&lt;-</span> LOQ_Mat[, <span class="fu">colnames</span>(target_demoData)]</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="fu">fData</span>(target_demoData)<span class="sc">$</span>DetectedSegments <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(LOQ_Mat, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="fu">fData</span>(target_demoData)<span class="sc">$</span>DetectionRate <span class="ot">&lt;-</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fData</span>(target_demoData)<span class="sc">$</span>DetectedSegments <span class="sc">/</span> <span class="fu">nrow</span>(<span class="fu">pData</span>(target_demoData))</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene of interest detection table</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>goi <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PDCD1"</span>, <span class="st">"CD274"</span>, <span class="st">"IFNG"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD68"</span>, <span class="st">"EPCAM"</span>,</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>         <span class="st">"KRT18"</span>, <span class="st">"NPHS1"</span>, <span class="st">"NPHS2"</span>, <span class="st">"CALB1"</span>, <span class="st">"CLDN8"</span>)</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>goi_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Gene =</span> goi,</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">Number =</span> <span class="fu">fData</span>(target_demoData)[goi, <span class="st">"DetectedSegments"</span>],</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">DetectionRate =</span> <span class="fu">percent</span>(<span class="fu">fData</span>(target_demoData)[goi, <span class="st">"DetectionRate"</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot detection rate:</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>plot_detect <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Freq =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">50</span>))</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>plot_detect<span class="sc">$</span>Number <span class="ot">&lt;-</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>),</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) {<span class="fu">sum</span>(<span class="fu">fData</span>(target_demoData)<span class="sc">$</span>DetectionRate <span class="sc">&gt;=</span> x)}))</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>plot_detect<span class="sc">$</span>Rate <span class="ot">&lt;-</span> plot_detect<span class="sc">$</span>Number <span class="sc">/</span> <span class="fu">nrow</span>(<span class="fu">fData</span>(target_demoData))</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(plot_detect) <span class="ot">&lt;-</span> plot_detect<span class="sc">$</span>Freq</span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_detect, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Freq), <span class="at">y =</span> Rate, <span class="at">fill =</span> Rate)) <span class="sc">+</span></span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">formatC</span>(Number, <span class="at">format =</span> <span class="st">"d"</span>, <span class="at">big.mark =</span> <span class="st">","</span>)),</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">vjust =</span> <span class="fl">1.6</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"orange2"</span>, <span class="at">mid =</span> <span class="st">"lightblue"</span>,</span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">high =</span> <span class="st">"dodgerblue3"</span>, <span class="at">midpoint =</span> <span class="fl">0.65</span>,</span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>                         <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))) <span class="sc">+</span></span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"% of Segments"</span>,</span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Genes Detected, % of Panel &gt; LOQ"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to target genes detected in at least 10% of the samples.</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   Also manually include the negative control probe, for downstream use</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>negativeProbefData <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">fData</span>(target_demoData), CodeClass <span class="sc">==</span> <span class="st">"Negative"</span>)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>neg_probes <span class="ot">&lt;-</span> <span class="fu">unique</span>(negativeProbefData<span class="sc">$</span>TargetName)</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>target_demoData <span class="ot">&lt;-</span> </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>    target_demoData[<span class="fu">fData</span>(target_demoData)<span class="sc">$</span>DetectionRate <span class="sc">&gt;=</span> <span class="fl">0.1</span> <span class="sc">|</span></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">fData</span>(target_demoData)<span class="sc">$</span>TargetName <span class="sc">%in%</span> neg_probes, ]</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(target_demoData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Features  Samples 
   10131      221 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># retain only detected genes of interest</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>goi <span class="ot">&lt;-</span> goi[goi <span class="sc">%in%</span> <span class="fu">rownames</span>(target_demoData)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)  <span class="co"># for melt</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)   <span class="co"># for plot_grid</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Graph Q3 value vs negGeoMean of Negatives</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>ann_of_interest <span class="ot">&lt;-</span> <span class="st">"region"</span></span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>Stat_data <span class="ot">&lt;-</span> </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">row.names =</span> <span class="fu">colnames</span>(<span class="fu">exprs</span>(target_demoData)),</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">Segment =</span> <span class="fu">colnames</span>(<span class="fu">exprs</span>(target_demoData)),</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">Annotation =</span> <span class="fu">pData</span>(target_demoData)[, ann_of_interest],</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">Q3 =</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(<span class="fu">exprs</span>(target_demoData), <span class="dv">2</span>,</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>                                 quantile, <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">NegProbe =</span> <span class="fu">exprs</span>(target_demoData)[neg_probes, ])</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>Stat_data_m <span class="ot">&lt;-</span> <span class="fu">melt</span>(Stat_data, <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">"Q3"</span>, <span class="st">"NegProbe"</span>),</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">variable.name =</span> <span class="st">"Statistic"</span>, <span class="at">value.name =</span> <span class="st">"Value"</span>)</span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Stat_data_m,</span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a>               <span class="fu">aes</span>(<span class="at">x =</span> Value, <span class="at">fill =</span> Statistic)) <span class="sc">+</span></span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">40</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Annotation, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="dv">3</span>, <span class="at">type =</span> <span class="st">"qual"</span>) <span class="sc">+</span></span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Counts"</span>, <span class="at">y =</span> <span class="st">"Segments, #"</span>)</span>
<span id="cb87-23"><a href="#cb87-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-24"><a href="#cb87-24" aria-hidden="true" tabindex="-1"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Stat_data,</span>
<span id="cb87-25"><a href="#cb87-25" aria-hidden="true" tabindex="-1"></a>               <span class="fu">aes</span>(<span class="at">x =</span> NegProbe, <span class="at">y =</span> Q3, <span class="at">color =</span> Annotation)) <span class="sc">+</span></span>
<span id="cb87-26"><a href="#cb87-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb87-27"><a href="#cb87-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">"none"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb87-28"><a href="#cb87-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span> </span>
<span id="cb87-29"><a href="#cb87-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb87-30"><a href="#cb87-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb87-31"><a href="#cb87-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Negative Probe GeoMean, Counts"</span>, <span class="at">y =</span> <span class="st">"Q3 Value, Counts"</span>)</span>
<span id="cb87-32"><a href="#cb87-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-33"><a href="#cb87-33" aria-hidden="true" tabindex="-1"></a>plt3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Stat_data,</span>
<span id="cb87-34"><a href="#cb87-34" aria-hidden="true" tabindex="-1"></a>               <span class="fu">aes</span>(<span class="at">x =</span> NegProbe, <span class="at">y =</span> Q3 <span class="sc">/</span> NegProbe, <span class="at">color =</span> Annotation)) <span class="sc">+</span></span>
<span id="cb87-35"><a href="#cb87-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb87-36"><a href="#cb87-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb87-37"><a href="#cb87-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span> </span>
<span id="cb87-38"><a href="#cb87-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb87-39"><a href="#cb87-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb87-40"><a href="#cb87-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Negative Probe GeoMean, Counts"</span>, <span class="at">y =</span> <span class="st">"Q3/NegProbe Value, Counts"</span>)</span>
<span id="cb87-41"><a href="#cb87-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-42"><a href="#cb87-42" aria-hidden="true" tabindex="-1"></a>btm_row <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(plt2, plt3, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"B"</span>, <span class="st">""</span>),</span>
<span id="cb87-43"><a href="#cb87-43" aria-hidden="true" tabindex="-1"></a>                     <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">0.43</span>,<span class="fl">0.57</span>))</span>
<span id="cb87-44"><a href="#cb87-44" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plt1, btm_row, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">""</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Q3 norm (75th percentile) for WTA/CTA  with or without custom spike-ins</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>target_demoData <span class="ot">&lt;-</span> <span class="fu">normalize</span>(target_demoData ,</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">norm_method =</span> <span class="st">"quant"</span>, </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">desiredQuantile =</span> .<span class="dv">75</span>,</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">toElt =</span> <span class="st">"q_norm"</span>)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Background normalization for WTA/CTA without custom spike-in</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>target_demoData <span class="ot">&lt;-</span> <span class="fu">normalize</span>(target_demoData ,</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>                             <span class="at">norm_method =</span> <span class="st">"neg"</span>, </span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>                             <span class="at">fromElt =</span> <span class="st">"exprs"</span>,</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                             <span class="at">toElt =</span> <span class="st">"neg_norm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">exprs</span>(target_demoData)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="st">"#9EDAE5"</span>, <span class="at">main =</span> <span class="st">"Raw Counts"</span>,</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">log =</span> <span class="st">"y"</span>, <span class="at">names =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">xlab =</span> <span class="st">"Segment"</span>,</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Counts, Raw"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">assayDataElement</span>(target_demoData[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], <span class="at">elt =</span> <span class="st">"q_norm"</span>),</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="st">"#2CA02C"</span>, <span class="at">main =</span> <span class="st">"Q3 Norm Counts"</span>,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">log =</span> <span class="st">"y"</span>, <span class="at">names =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">xlab =</span> <span class="st">"Segment"</span>,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Counts, Q3 Normalized"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">assayDataElement</span>(target_demoData[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], <span class="at">elt =</span> <span class="st">"neg_norm"</span>),</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="st">"#FF7F0E"</span>, <span class="at">main =</span> <span class="st">"Neg Norm Counts"</span>,</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">log =</span> <span class="st">"y"</span>, <span class="at">names =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">xlab =</span> <span class="st">"Segment"</span>,</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Counts, Neg. Normalized"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(umap)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rtsne)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="co"># update defaults for umap to contain a stable random_state (seed)</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>custom_umap <span class="ot">&lt;-</span> umap<span class="sc">::</span>umap.defaults</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>custom_umap<span class="sc">$</span>random_state <span class="ot">&lt;-</span> <span class="dv">42</span></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a><span class="co"># run UMAP</span></span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>umap_out <span class="ot">&lt;-</span></span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">umap</span>(<span class="fu">t</span>(<span class="fu">log2</span>(<span class="fu">assayDataElement</span>(target_demoData , <span class="at">elt =</span> <span class="st">"q_norm"</span>))),  </span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">config =</span> custom_umap)</span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)[, <span class="fu">c</span>(<span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span>)] <span class="ot">&lt;-</span> umap_out<span class="sc">$</span>layout[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">pData</span>(target_demoData),</span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> UMAP1, <span class="at">y =</span> UMAP2, <span class="at">color =</span> region, <span class="at">shape =</span> class)) <span class="sc">+</span></span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run tSNE</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>) <span class="co"># set the seed for tSNE as well</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>tsne_out <span class="ot">&lt;-</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Rtsne</span>(<span class="fu">t</span>(<span class="fu">log2</span>(<span class="fu">assayDataElement</span>(target_demoData , <span class="at">elt =</span> <span class="st">"q_norm"</span>))),</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">perplexity =</span> <span class="fu">ncol</span>(target_demoData)<span class="sc">*</span>.<span class="dv">15</span>)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)[, <span class="fu">c</span>(<span class="st">"tSNE1"</span>, <span class="st">"tSNE2"</span>)] <span class="ot">&lt;-</span> tsne_out<span class="sc">$</span>Y[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">pData</span>(target_demoData),</span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> tSNE1, <span class="at">y =</span> tSNE2, <span class="at">color =</span> region, <span class="at">shape =</span> class)) <span class="sc">+</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)  <span class="co"># for pheatmap</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co"># create a log2 transform of the data for analysis</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="fu">assayDataElement</span>(<span class="at">object =</span> target_demoData, <span class="at">elt =</span> <span class="st">"log_q"</span>) <span class="ot">&lt;-</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assayDataApply</span>(target_demoData, <span class="dv">2</span>, <span class="at">FUN =</span> log, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">elt =</span> <span class="st">"q_norm"</span>)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a><span class="co"># create CV function</span></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>calc_CV <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {<span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">mean</span>(x)}</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>CV_dat <span class="ot">&lt;-</span> <span class="fu">assayDataApply</span>(target_demoData,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">elt =</span> <span class="st">"log_q"</span>, <span class="at">MARGIN =</span> <span class="dv">1</span>, calc_CV)</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a><span class="co"># show the highest CD genes and their CV values</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(CV_dat, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  CAMK2N1    AKR1C1      AQP2     GDF15       REN 
0.5886006 0.5114973 0.4607206 0.4196469 0.4193216 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   CAMK2N1    AKR1C1      AQP2     GDF15       REN </span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.5886006 0.5114973 0.4607206 0.4196469 0.4193216</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify genes in the top 3rd of the CV values</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>GOI <span class="ot">&lt;-</span> <span class="fu">names</span>(CV_dat)[CV_dat <span class="sc">&gt;</span> <span class="fu">quantile</span>(CV_dat, <span class="fl">0.8</span>)]</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">assayDataElement</span>(target_demoData[GOI, ], <span class="at">elt =</span> <span class="st">"log_q"</span>),</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="cn">NA</span>,</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">"average"</span>,</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> <span class="st">"correlation"</span>,</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>, <span class="fl">0.05</span>),</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"purple3"</span>, <span class="st">"black"</span>, <span class="st">"yellow2"</span>))(<span class="dv">120</span>),</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> </span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">pData</span>(target_demoData)[, <span class="fu">c</span>(<span class="st">"class"</span>, <span class="st">"segment"</span>, <span class="st">"region"</span>)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert test variables to factors</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>testRegion <span class="ot">&lt;-</span> </span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(<span class="fu">pData</span>(target_demoData)<span class="sc">$</span>region, <span class="fu">c</span>(<span class="st">"glomerulus"</span>, <span class="st">"tubule"</span>))</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)[[<span class="st">"slide"</span>]] <span class="ot">&lt;-</span> </span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(<span class="fu">pData</span>(target_demoData)[[<span class="st">"slide name"</span>]])</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a><span class="fu">assayDataElement</span>(<span class="at">object =</span> target_demoData, <span class="at">elt =</span> <span class="st">"log_q"</span>) <span class="ot">&lt;-</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assayDataApply</span>(target_demoData, <span class="dv">2</span>, <span class="at">FUN =</span> log, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">elt =</span> <span class="st">"q_norm"</span>)</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(status <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"DKD"</span>, <span class="st">"normal"</span>)) {</span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>    ind <span class="ot">&lt;-</span> <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>class <span class="sc">==</span> status</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>    mixedOutmc <span class="ot">&lt;-</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mixedModelDE</span>(target_demoData[, ind],</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">elt =</span> <span class="st">"log_q"</span>,</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">modelFormula =</span> <span class="sc">~</span> testRegion <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">+</span> testRegion <span class="sc">|</span> slide),</span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">groupVar =</span> <span class="st">"testRegion"</span>,</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">nCores =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>(),</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiCore =</span> <span class="cn">FALSE</span>)</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># format results as data.frame</span></span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, mixedOutmc[<span class="st">"lsmeans"</span>, ])</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a>    tests <span class="ot">&lt;-</span> <span class="fu">rownames</span>(r_test)</span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(r_test)</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Contrast <span class="ot">&lt;-</span> tests</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Gene <span class="ot">&lt;-</span> </span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">colnames</span>(mixedOutmc),</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>                      rep, <span class="fu">nrow</span>(mixedOutmc[<span class="st">"lsmeans"</span>, ][[<span class="dv">1</span>]])))</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Subset <span class="ot">&lt;-</span> status</span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(r_test<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span>, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> r_test[, <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"Subset"</span>, <span class="st">"Contrast"</span>, <span class="st">"Estimate"</span>, </span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Pr(&gt;|t|)"</span>, <span class="st">"FDR"</span>)]</span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results, r_test)</span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">subset</span>(results, Gene <span class="sc">%in%</span> goi <span class="sc">&amp;</span> Subset <span class="sc">==</span> <span class="st">"normal"</span>), <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">caption =</span> <span class="st">"DE results for Genes of Interest"</span>,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">align =</span> <span class="st">"lc"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>DE results for Genes of Interest</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 11%">
<col style="width: 28%">
<col style="width: 14%">
<col style="width: 27%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: center;">Subset</th>
<th style="text-align: left;">Contrast</th>
<th style="text-align: center;">Estimate</th>
<th style="text-align: left;">Pr(&gt;|t|)</th>
<th style="text-align: center;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">KRT18</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-1.169</td>
<td style="text-align: left;">0.076</td>
<td style="text-align: center;">0.237</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD68</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-0.153</td>
<td style="text-align: left;">0.289</td>
<td style="text-align: center;">0.483</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8A</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-0.227</td>
<td style="text-align: left;">0.147</td>
<td style="text-align: center;">0.332</td>
</tr>
<tr class="even">
<td style="text-align: left;">NPHS1</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">3.809</td>
<td style="text-align: left;">0.001</td>
<td style="text-align: center;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CALB1</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-2.014</td>
<td style="text-align: left;">0.027</td>
<td style="text-align: center;">0.138</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD274</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">0.223</td>
<td style="text-align: left;">0.031</td>
<td style="text-align: center;">0.147</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NPHS2</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">5.430</td>
<td style="text-align: left;">0.002</td>
<td style="text-align: center;">0.025</td>
</tr>
<tr class="even">
<td style="text-align: left;">CLDN8</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-1.961</td>
<td style="text-align: left;">0.001</td>
<td style="text-align: center;">0.011</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EPCAM</td>
<td style="text-align: center;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: center;">-2.297</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: center;">0.006</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert test variables to factors</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(target_demoData)<span class="sc">$</span>testClass <span class="ot">&lt;-</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(<span class="fu">pData</span>(target_demoData)<span class="sc">$</span>class, <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"DKD"</span>))</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="co"># run LMM:</span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="co"># formula follows conventions defined by the lme4 package</span></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>results2 <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(region <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"glomerulus"</span>, <span class="st">"tubule"</span>)) {</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>    ind <span class="ot">&lt;-</span> <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>region <span class="sc">==</span> region</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>    mixedOutmc <span class="ot">&lt;-</span></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mixedModelDE</span>(target_demoData[, ind],</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">elt =</span> <span class="st">"log_q"</span>,</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">modelFormula =</span> <span class="sc">~</span> testClass <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> slide),</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">groupVar =</span> <span class="st">"testClass"</span>,</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">nCores =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>(),</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiCore =</span> <span class="cn">FALSE</span>)</span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># format results as data.frame</span></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, mixedOutmc[<span class="st">"lsmeans"</span>, ])</span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a>    tests <span class="ot">&lt;-</span> <span class="fu">rownames</span>(r_test)</span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(r_test)</span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Contrast <span class="ot">&lt;-</span> tests</span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># use lapply in case you have multiple levels of your test factor to</span></span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># correctly associate gene name with it's row in the results table</span></span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Gene <span class="ot">&lt;-</span> </span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">colnames</span>(mixedOutmc),</span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a>                      rep, <span class="fu">nrow</span>(mixedOutmc[<span class="st">"lsmeans"</span>, ][[<span class="dv">1</span>]])))</span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>Subset <span class="ot">&lt;-</span> region</span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a>    r_test<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(r_test<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span>, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a>    r_test <span class="ot">&lt;-</span> r_test[, <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"Subset"</span>, <span class="st">"Contrast"</span>, <span class="st">"Estimate"</span>, </span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Pr(&gt;|t|)"</span>, <span class="st">"FDR"</span>)]</span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a>    results2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results2, r_test)</span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">subset</span>(results2, Gene <span class="sc">%in%</span> goi <span class="sc">&amp;</span> Subset <span class="sc">==</span> <span class="st">"tubule"</span>), <span class="at">digits =</span> <span class="dv">3</span>,</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">caption =</span> <span class="st">"DE results for Genes of Interest"</span>,</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">align =</span> <span class="st">"lc"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>DE results for Genes of Interest</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: center;">Subset</th>
<th style="text-align: left;">Contrast</th>
<th style="text-align: center;">Estimate</th>
<th style="text-align: left;">Pr(&gt;|t|)</th>
<th style="text-align: center;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">KRT18</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.096</td>
<td style="text-align: left;">0.748</td>
<td style="text-align: center;">0.997</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD68</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.726</td>
<td style="text-align: left;">0.124</td>
<td style="text-align: center;">0.965</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CD8A</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.057</td>
<td style="text-align: left;">0.826</td>
<td style="text-align: center;">0.998</td>
</tr>
<tr class="even">
<td style="text-align: left;">NPHS1</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.131</td>
<td style="text-align: left;">0.624</td>
<td style="text-align: center;">0.997</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CALB1</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">1.408</td>
<td style="text-align: left;">0.013</td>
<td style="text-align: center;">0.652</td>
</tr>
<tr class="even">
<td style="text-align: left;">CD274</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.252</td>
<td style="text-align: left;">0.522</td>
<td style="text-align: center;">0.997</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NPHS2</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">-0.128</td>
<td style="text-align: left;">0.730</td>
<td style="text-align: center;">0.997</td>
</tr>
<tr class="even">
<td style="text-align: left;">CLDN8</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">0.979</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: center;">0.058</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EPCAM</td>
<td style="text-align: center;">tubule</td>
<td style="text-align: left;">normal - DKD</td>
<td style="text-align: center;">0.339</td>
<td style="text-align: left;">0.382</td>
<td style="text-align: center;">0.997</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel) </span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Categorize Results based on P-value &amp; FDR for plotting</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color <span class="ot">&lt;-</span> <span class="st">"NS or FC &lt; 0.5"</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color[results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"P &lt; 0.05"</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color[results<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"FDR &lt; 0.05"</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color[results<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.001</span>] <span class="ot">&lt;-</span> <span class="st">"FDR &lt; 0.001"</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color[<span class="fu">abs</span>(results<span class="sc">$</span>Estimate) <span class="sc">&lt;</span> <span class="fl">0.5</span>] <span class="ot">&lt;-</span> <span class="st">"NS or FC &lt; 0.5"</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>Color <span class="ot">&lt;-</span> <span class="fu">factor</span>(results<span class="sc">$</span>Color,</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"NS or FC &lt; 0.5"</span>, <span class="st">"P &lt; 0.05"</span>,</span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"FDR &lt; 0.05"</span>, <span class="st">"FDR &lt; 0.001"</span>))</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co"># pick top genes for either side of volcano to label</span></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a><span class="co"># order genes for convenience:</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>invert_P <span class="ot">&lt;-</span> (<span class="sc">-</span><span class="fu">log10</span>(results<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span>)) <span class="sc">*</span> <span class="fu">sign</span>(results<span class="sc">$</span>Estimate)</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a>top_g <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(cond <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"DKD"</span>, <span class="st">"normal"</span>)) {</span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a>    ind <span class="ot">&lt;-</span> results<span class="sc">$</span>Subset <span class="sc">==</span> cond</span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a>    top_g <span class="ot">&lt;-</span> <span class="fu">c</span>(top_g,</span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a>               results[ind, <span class="st">'Gene'</span>][</span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">order</span>(results[ind, <span class="st">'invert_P'</span>], <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>]],</span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a>               results[ind, <span class="st">'Gene'</span>][</span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">order</span>(results[ind, <span class="st">'invert_P'</span>], <span class="at">decreasing =</span> <span class="cn">FALSE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>]])</span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a>top_g <span class="ot">&lt;-</span> <span class="fu">unique</span>(top_g)</span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results[, <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span><span class="fu">ncol</span>(results)] <span class="co"># remove invert_P from matrix</span></span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Graph results</span></span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(results,</span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> Estimate, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span>),</span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Color, <span class="at">label =</span> Gene)) <span class="sc">+</span></span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="sc">-</span><span class="fl">0.5</span>), <span class="at">lty =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">lty =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Enriched in Tubules &lt;- log2(FC) -&gt; Enriched in Glomeruli"</span>,</span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Significance, -log10(P)"</span>,</span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">"Significance"</span>) <span class="sc">+</span></span>
<span id="cb101-37"><a href="#cb101-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">FDR &lt; 0.001</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"dodgerblue"</span>,</span>
<span id="cb101-38"><a href="#cb101-38" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">FDR &lt; 0.05</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"lightblue"</span>,</span>
<span id="cb101-39"><a href="#cb101-39" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">P &lt; 0.05</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"orange2"</span>,</span>
<span id="cb101-40"><a href="#cb101-40" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">NS or FC &lt; 0.5</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"gray"</span>),</span>
<span id="cb101-41"><a href="#cb101-41" aria-hidden="true" tabindex="-1"></a>                       <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">4</span>))) <span class="sc">+</span></span>
<span id="cb101-42"><a href="#cb101-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.05</span>))) <span class="sc">+</span></span>
<span id="cb101-43"><a href="#cb101-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">subset</span>(results, Gene <span class="sc">%in%</span> top_g <span class="sc">&amp;</span> FDR <span class="sc">&lt;</span> <span class="fl">0.001</span>),</span>
<span id="cb101-44"><a href="#cb101-44" aria-hidden="true" tabindex="-1"></a>                    <span class="at">size =</span> <span class="dv">4</span>, <span class="at">point.padding =</span> <span class="fl">0.15</span>, <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb101-45"><a href="#cb101-45" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.segment.length =</span> .<span class="dv">1</span>, <span class="at">box.padding =</span> .<span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>,</span>
<span id="cb101-46"><a href="#cb101-46" aria-hidden="true" tabindex="-1"></a>                    <span class="at">max.overlaps =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb101-47"><a href="#cb101-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb101-48"><a href="#cb101-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb101-49"><a href="#cb101-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Subset, <span class="at">scales =</span> <span class="st">"free_y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Duplicated aesthetics after name standardisation: size</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">subset</span>(results, Gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'PDHA1'</span>,<span class="st">'ITGB1'</span>)), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 7%">
<col style="width: 8%">
<col style="width: 23%">
<col style="width: 12%">
<col style="width: 22%">
<col style="width: 11%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Gene</th>
<th style="text-align: left;">Subset</th>
<th style="text-align: left;">Contrast</th>
<th style="text-align: right;">Estimate</th>
<th style="text-align: right;">Pr(&gt;|t|)</th>
<th style="text-align: right;">FDR</th>
<th style="text-align: left;">Color</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">PDHA1</td>
<td style="text-align: left;">DKD</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: right;">-1.0657650</td>
<td style="text-align: right;">0.0065159</td>
<td style="text-align: right;">0.1122655</td>
<td style="text-align: left;">P &lt; 0.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">ITGB1</td>
<td style="text-align: left;">DKD</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: right;">0.6769474</td>
<td style="text-align: right;">0.0307688</td>
<td style="text-align: right;">0.2192658</td>
<td style="text-align: left;">P &lt; 0.05</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PDHA1</td>
<td style="text-align: left;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: right;">-1.5416314</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">FDR &lt; 0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">ITGB1</td>
<td style="text-align: left;">normal</td>
<td style="text-align: left;">glomerulus - tubule</td>
<td style="text-align: right;">1.5042831</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">FDR &lt; 0.001</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># show expression for a single target: PDHA1</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">pData</span>(target_demoData),</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> region, <span class="at">fill =</span> region,</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">assayDataElement</span>(target_demoData[<span class="st">"PDHA1"</span>, ],</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>                                <span class="at">elt =</span> <span class="st">"q_norm"</span>))) <span class="sc">+</span></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">width =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"PDHA1 Expression"</span>) <span class="sc">+</span></span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>class) <span class="sc">+</span></span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>glom <span class="ot">&lt;-</span> <span class="fu">pData</span>(target_demoData)<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"glomerulus"</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># show expression of PDHA1 vs ITGB1</span></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">pData</span>(target_demoData),</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">assayDataElement</span>(target_demoData[<span class="st">"PDHA1"</span>, ],</span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">elt =</span> <span class="st">"q_norm"</span>),</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">assayDataElement</span>(target_demoData[<span class="st">"ITGB1"</span>, ],</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">elt =</span> <span class="st">"q_norm"</span>),</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> region)) <span class="sc">+</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span></span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">max</span>(<span class="fu">assayDataElement</span>(target_demoData[<span class="st">"PDHA1"</span>, glom],</span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">elt =</span> <span class="st">"q_norm"</span>)),</span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">lty =</span> <span class="st">"dashed"</span>, <span class="at">col =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span></span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">max</span>(<span class="fu">assayDataElement</span>(target_demoData[<span class="st">"ITGB1"</span>, <span class="sc">!</span>glom],</span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">elt =</span> <span class="st">"q_norm"</span>)),</span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">lty =</span> <span class="st">"dashed"</span>, <span class="at">col =</span> <span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb105-19"><a href="#cb105-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb105-20"><a href="#cb105-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span> </span>
<span id="cb105-21"><a href="#cb105-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb105-22"><a href="#cb105-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PDHA1 Expression"</span>, <span class="at">y =</span> <span class="st">"ITGB1 Expression"</span>) <span class="sc">+</span></span>
<span id="cb105-23"><a href="#cb105-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># select top significant genes based on significance, plot with pheatmap</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>GOI <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">subset</span>(results, <span class="st">`</span><span class="at">FDR</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.001</span>)<span class="sc">$</span>Gene)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">log2</span>(<span class="fu">assayDataElement</span>(target_demoData[GOI, ], <span class="at">elt =</span> <span class="st">"q_norm"</span>)),</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="cn">NA</span>,</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">"average"</span>,</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> <span class="st">"correlation"</span>,</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">cutree_cols =</span> <span class="dv">2</span>, <span class="at">cutree_rows =</span> <span class="dv">2</span>,</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>, <span class="fl">0.05</span>),</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"purple3"</span>, <span class="st">"black"</span>, <span class="st">"yellow2"</span>))(<span class="dv">120</span>),</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> <span class="fu">pData</span>(target_demoData)[, <span class="fu">c</span>(<span class="st">"region"</span>, <span class="st">"class"</span>)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>MeanExp <span class="ot">&lt;-</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowMeans</span>(<span class="fu">assayDataElement</span>(target_demoData,</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">elt =</span> <span class="st">"q_norm"</span>))</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>top_g2 <span class="ot">&lt;-</span> results<span class="sc">$</span>Gene[results<span class="sc">$</span>Gene <span class="sc">%in%</span> top_g <span class="sc">&amp;</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>                           results<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.001</span> <span class="sc">&amp;</span></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">abs</span>(results<span class="sc">$</span>Estimate) <span class="sc">&gt;</span> .<span class="dv">5</span> <span class="sc">&amp;</span></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>                           results<span class="sc">$</span>MeanExp <span class="sc">&gt;</span> <span class="fu">quantile</span>(results<span class="sc">$</span>MeanExp, <span class="fl">0.9</span>)]</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">subset</span>(results, <span class="sc">!</span>Gene <span class="sc">%in%</span> neg_probes),</span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> MeanExp, <span class="at">y =</span> Estimate,</span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="st">`</span><span class="at">Pr(&gt;|t|)</span><span class="st">`</span>),</span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> Color, <span class="at">label =</span> Gene)) <span class="sc">+</span></span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="sc">-</span><span class="fl">0.5</span>), <span class="at">lty =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Enriched in Glomeruli &lt;- log2(FC) -&gt; Enriched in Tubules"</span>,</span>
<span id="cb107-18"><a href="#cb107-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Mean Expression"</span>,</span>
<span id="cb107-19"><a href="#cb107-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">"Significance"</span>) <span class="sc">+</span></span>
<span id="cb107-20"><a href="#cb107-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">FDR &lt; 0.001</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"dodgerblue"</span>,</span>
<span id="cb107-21"><a href="#cb107-21" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">FDR &lt; 0.05</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"lightblue"</span>,</span>
<span id="cb107-22"><a href="#cb107-22" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">P &lt; 0.05</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"orange2"</span>,</span>
<span id="cb107-23"><a href="#cb107-23" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">`</span><span class="at">NS or FC &lt; 0.5</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"gray"</span>)) <span class="sc">+</span></span>
<span id="cb107-24"><a href="#cb107-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">subset</span>(results, Gene <span class="sc">%in%</span> top_g2),</span>
<span id="cb107-25"><a href="#cb107-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">size =</span> <span class="dv">4</span>, <span class="at">point.padding =</span> <span class="fl">0.15</span>, <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb107-26"><a href="#cb107-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.segment.length =</span> .<span class="dv">1</span>, <span class="at">box.padding =</span> .<span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb107-27"><a href="#cb107-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb107-28"><a href="#cb107-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Subset, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Duplicated aesthetics after name standardisation: size</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="GEO_Project_code_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.2           pheatmap_1.0.12         Rtsne_0.16             
 [4] umap_0.2.9.0            cowplot_1.1.1           reshape2_1.4.4         
 [7] scales_1.2.1            ggforce_0.4.1           dplyr_1.0.10           
[10] knitr_1.40              GeoMxWorkflows_1.5.0    GeomxTools_3.2.0       
[13] NanoStringNCTools_1.6.0 ggplot2_3.4.0           S4Vectors_0.36.0       
[16] Biobase_2.58.0          BiocGenerics_0.44.0     BiocManager_1.30.19    

loaded via a namespace (and not attached):
 [1] nlme_3.1-160           bitops_1.0-7           EnvStats_2.7.0        
 [4] RColorBrewer_1.1-3     GenomeInfoDb_1.34.3    numDeriv_2016.8-1.1   
 [7] tools_4.2.2            utf8_1.2.2             R6_2.5.1              
[10] vipor_0.4.5            DBI_1.1.3              colorspace_2.0-3      
[13] withr_2.5.0            sp_1.5-1               tidyselect_1.2.0      
[16] GGally_2.1.2           compiler_4.2.2         progressr_0.11.0      
[19] cli_3.4.1              labeling_0.4.2         askpass_1.1           
[22] systemfonts_1.0.4      stringr_1.4.1          digest_0.6.30         
[25] minqa_1.2.5            rmarkdown_2.18         XVector_0.38.0        
[28] pkgconfig_2.0.3        htmltools_0.5.3        parallelly_1.32.1     
[31] lme4_1.1-31            highr_0.9              fastmap_1.1.0         
[34] htmlwidgets_1.5.4      rlang_1.0.6            ggthemes_4.2.4        
[37] readxl_1.4.1           rstudioapi_0.14        farver_2.1.1          
[40] generics_0.1.3         jsonlite_1.8.3         RCurl_1.98-1.9        
[43] magrittr_2.0.3         GenomeInfoDbData_1.2.9 Matrix_1.5-3          
[46] Rcpp_1.0.9             ggbeeswarm_0.6.0       munsell_0.5.0         
[49] fansi_1.0.3            reticulate_1.26        lifecycle_1.0.3       
[52] stringi_1.7.8          yaml_2.3.6             MASS_7.3-58.1         
[55] zlibbioc_1.44.0        plyr_1.8.8             grid_4.2.2            
[58] parallel_4.2.2         listenv_0.8.0          crayon_1.5.2          
[61] lattice_0.20-45        Biostrings_2.66.0      splines_4.2.2         
[64] pillar_1.8.1           uuid_1.1-0             boot_1.3-28           
[67] rjson_0.2.21           future.apply_1.10.0    codetools_0.2-18      
[70] glue_1.6.2             ggiraph_0.8.3          evaluate_0.18         
[73] outliers_0.15          SeuratObject_4.1.3     data.table_1.14.6     
[76] png_0.1-7              tweenr_2.0.2           vctrs_0.5.1           
[79] nloptr_2.0.3           cellranger_1.1.0       openssl_2.0.4         
[82] polyclip_1.10-4        gtable_0.3.1           purrr_0.3.5           
[85] reshape_0.8.9          future_1.29.0          assertthat_0.2.1      
[88] xfun_0.35              RSpectra_0.16-1        tibble_3.1.8          
[91] lmerTest_3.1-3         beeswarm_0.4.0         IRanges_2.32.0        
[94] globals_0.16.1         BiocStyle_2.26.0      </code></pre>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>