<!DOCTYPE html>
<html lang="en"><head>
<script src="geomx_presentation_files/libs/clipboard/clipboard.min.js"></script>
<script src="geomx_presentation_files/libs/quarto-html/tabby.min.js"></script>
<script src="geomx_presentation_files/libs/quarto-html/popper.min.js"></script>
<script src="geomx_presentation_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="geomx_presentation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="geomx_presentation_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="geomx_presentation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.1.163">

  <meta name="author" content="Jason Reeves, Prajan Divakar, Nicole Ortogero, Maddy Griswold, Zhi Yang, Stephanie Zimmerman, Rona Vitancol and David Henderson">
  <title>Analyzing GeoMx-NGS RNA Expression Data with GeomxTools</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="geomx_presentation_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="geomx_presentation_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="geomx_presentation_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="geomx_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="geomx_presentation_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="geomx_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="geomx_presentation_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="center">
  <h1 class="title">Analyzing GeoMx-NGS RNA Expression Data with GeomxTools</h1>
  <p class="author">Jason Reeves, Prajan Divakar, Nicole Ortogero, Maddy Griswold, Zhi Yang, Stephanie Zimmerman, Rona Vitancol and David Henderson</p>
</section>

<section id="overview-of-geomx-ngs-workflow" class="slide level2">
<h2>Overview of GeoMx-NGS Workflow</h2>

<img data-src="figures/GeoMx-NGS.png" class="r-stretch"></section>
<section id="overview-of-the-dataset" class="slide level2">
<h2>Overview of the Dataset</h2>
<ul>
<li>We will analyze a GeoMx kidney dataset created with the human whole transcriptome atlas (WTA) assay. The dataset includes 4 diabetic kidney disease (DKD) and 3 healthy kidney tissue samples.</li>
</ul>
</section>
<section id="overview-of-the-dataset-1" class="slide level2">
<h2>Overview of the Dataset</h2>
<ul>
<li><p>The key data files are:</p>
<ul>
<li><p>DCCs files - expression count data and sequencing quality metadata</p></li>
<li><p>PKCs file(s) - probe assay metadata describing the gene targets present in the data</p></li>
<li><p>Annotation file - useful tissue information, including the type of segment profiled (ex: glomerulus vs.&nbsp;tubule), segment area/nuclei count, and other tissue characteristics (ex: diseased vs.&nbsp;healthy).</p></li>
</ul></li>
</ul>
</section>
<section id="overview-of-the-dataset-2" class="slide level2">
<h2>Overview of the Dataset</h2>

<img data-src="figures/GeoMxSet_Object_Overview.png" class="r-stretch"></section>
<section id="overview-of-the-analysis-methods" class="slide level2">
<h2>Overview of the Analysis Methods</h2>

<img data-src="figures/QCandPreprocessing.png" class="r-stretch"></section>
<section id="overview-of-the-tools" class="slide level2">
<h2>Overview of the Tools</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb1-1"><a href="#cb1-1"></a>library(NanoStringNCTools)</span>
<span id="cb1-2"><a href="#cb1-2"></a>library(GeomxTools)</span>
<span id="cb1-3"><a href="#cb1-3"></a>library(GeoMxWorkflows)</span>
<span id="cb1-4"><a href="#cb1-4"></a>library(knitr)</span>
<span id="cb1-5"><a href="#cb1-5"></a>library(ggplot2)</span>
<span id="cb1-6"><a href="#cb1-6"></a>library(scales) # for percent</span>
<span id="cb1-7"><a href="#cb1-7"></a>library(reshape2)  # for melt</span>
<span id="cb1-8"><a href="#cb1-8"></a>library(cowplot)   # for plot_grid</span>
<span id="cb1-9"><a href="#cb1-9"></a>library(umap)</span>
<span id="cb1-10"><a href="#cb1-10"></a>library(Rtsne)</span>
<span id="cb1-11"><a href="#cb1-11"></a>library(pheatmap)  # for heatmap</span>
<span id="cb1-12"><a href="#cb1-12"></a>library(ggrepel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="take-home-highlights" class="slide level2">
<h2>Take Home Highlights</h2>
<ul>
<li>QC &amp; Pre-processing and Normalization</li>
<li>Unsupervised Analysis
<ul>
<li>UMAP &amp; t-SNE plots</li>
<li>Clustering high coefficient of variation (CV) genes (heatmap)</li>
</ul></li>
<li>Visualizing Differential Expression
<ul>
<li>Volcano Plot</li>
<li>MA plot</li>
<li>heatmap</li>
</ul></li>
</ul>
</section>
<section id="selected-interesting-results" class="slide level2">
<h2>Selected Interesting Results</h2>
</section>
<section id="qc-pre-processing" class="slide level2">
<h2>QC &amp; Pre-processing</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb2-1"><a href="#cb2-1"></a>col_by &lt;- "segment"</span>
<span id="cb2-2"><a href="#cb2-2"></a># Graphical summaries of QC statistics plot function</span>
<span id="cb2-3"><a href="#cb2-3"></a>QC_histogram &lt;- function(assay_data = NULL,</span>
<span id="cb2-4"><a href="#cb2-4"></a>                         annotation = NULL,</span>
<span id="cb2-5"><a href="#cb2-5"></a>                         fill_by = NULL,</span>
<span id="cb2-6"><a href="#cb2-6"></a>                         thr = NULL,</span>
<span id="cb2-7"><a href="#cb2-7"></a>                         scale_trans = NULL) {</span>
<span id="cb2-8"><a href="#cb2-8"></a>    plt &lt;- ggplot(assay_data,</span>
<span id="cb2-9"><a href="#cb2-9"></a>                  aes_string(x = paste0("unlist(`", annotation, "`)"),</span>
<span id="cb2-10"><a href="#cb2-10"></a>                             fill = fill_by)) +</span>
<span id="cb2-11"><a href="#cb2-11"></a>        geom_histogram(bins = 50) +</span>
<span id="cb2-12"><a href="#cb2-12"></a>        geom_vline(xintercept = thr, lty = "dashed", color = "black") +</span>
<span id="cb2-13"><a href="#cb2-13"></a>        theme_bw() + guides(fill = "none") +</span>
<span id="cb2-14"><a href="#cb2-14"></a>        facet_wrap(as.formula(paste("~", fill_by)), nrow = 4) +</span>
<span id="cb2-15"><a href="#cb2-15"></a>        labs(x = annotation, y = "Segments, #", title = annotation)</span>
<span id="cb2-16"><a href="#cb2-16"></a>    if(!is.null(scale_trans)) {</span>
<span id="cb2-17"><a href="#cb2-17"></a>        plt &lt;- plt +</span>
<span id="cb2-18"><a href="#cb2-18"></a>            scale_x_continuous(trans = scale_trans)</span>
<span id="cb2-19"><a href="#cb2-19"></a>    }</span>
<span id="cb2-20"><a href="#cb2-20"></a>    plt</span>
<span id="cb2-21"><a href="#cb2-21"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="qc-pre-processing-1" class="slide level2">
<h2>QC &amp; Pre-processing</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource {{r}}} number-lines code-with-copy"><code class="sourceCode"><span id="cb3-1"><a href="#cb3-1"></a>QC_histogram(sData(demoData), "Trimmed (%)", col_by, 80)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<img data-src="figures/unnamed-chunk-11-1.png" class="r-stretch"></section>
<section id="qc-pre-processing-2" class="slide level2">
<h2>QC &amp; Pre-processing</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode numberSource {{r}}} number-lines code-with-copy"><code class="sourceCode"><span id="cb4-1"><a href="#cb4-1"></a># Plot detection rate:</span>
<span id="cb4-2"><a href="#cb4-2"></a>plot_detect &lt;- data.frame(Freq = c(1, 5, 10, 20, 30, 50))</span>
<span id="cb4-3"><a href="#cb4-3"></a>plot_detect$Number &lt;-</span>
<span id="cb4-4"><a href="#cb4-4"></a>    unlist(lapply(c(0.01, 0.05, 0.1, 0.2, 0.3, 0.5),</span>
<span id="cb4-5"><a href="#cb4-5"></a>                  function(x) {sum(fData(target_demoData)$DetectionRate &gt;= x)}))</span>
<span id="cb4-6"><a href="#cb4-6"></a>plot_detect$Rate &lt;- plot_detect$Number / nrow(fData(target_demoData))</span>
<span id="cb4-7"><a href="#cb4-7"></a>rownames(plot_detect) &lt;- plot_detect$Freq</span>
<span id="cb4-8"><a href="#cb4-8"></a></span>
<span id="cb4-9"><a href="#cb4-9"></a>ggplot(plot_detect, aes(x = as.factor(Freq), y = Rate, fill = Rate)) +</span>
<span id="cb4-10"><a href="#cb4-10"></a>    geom_bar(stat = "identity") +</span>
<span id="cb4-11"><a href="#cb4-11"></a>    geom_text(aes(label = formatC(Number, format = "d", big.mark = ",")),</span>
<span id="cb4-12"><a href="#cb4-12"></a>              vjust = 1.6, color = "black", size = 4) +</span>
<span id="cb4-13"><a href="#cb4-13"></a>    scale_fill_gradient2(low = "orange2", mid = "lightblue",</span>
<span id="cb4-14"><a href="#cb4-14"></a>                         high = "dodgerblue3", midpoint = 0.65,</span>
<span id="cb4-15"><a href="#cb4-15"></a>                         limits = c(0,1),</span>
<span id="cb4-16"><a href="#cb4-16"></a>                         labels = scales::percent) +</span>
<span id="cb4-17"><a href="#cb4-17"></a>    theme_bw() +</span>
<span id="cb4-18"><a href="#cb4-18"></a>    scale_y_continuous(labels = scales::percent, limits = c(0,1),</span>
<span id="cb4-19"><a href="#cb4-19"></a>                       expand = expansion(mult = c(0, 0))) +</span>
<span id="cb4-20"><a href="#cb4-20"></a>    labs(x = "% of Segments",</span>
<span id="cb4-21"><a href="#cb4-21"></a>         y = "Genes Detected, % of Panel &gt; LOQ")</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="qc-pre-processing-3" class="slide level2">
<h2>QC &amp; Pre-processing</h2>

<img data-src="figures/unnamed-chunk-30-1.png" class="r-stretch"></section>
<section id="normalization" class="slide level2">
<h2>Normalization</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb5-1"><a href="#cb5-1"></a>library(reshape2)  # for melt</span>
<span id="cb5-2"><a href="#cb5-2"></a>library(cowplot)   # for plot_grid</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a># Graph Q3 value vs negGeoMean of Negatives</span>
<span id="cb5-5"><a href="#cb5-5"></a>ann_of_interest &lt;- "region"</span>
<span id="cb5-6"><a href="#cb5-6"></a>Stat_data &lt;- </span>
<span id="cb5-7"><a href="#cb5-7"></a>    data.frame(row.names = colnames(exprs(target_demoData)),</span>
<span id="cb5-8"><a href="#cb5-8"></a>               Segment = colnames(exprs(target_demoData)),</span>
<span id="cb5-9"><a href="#cb5-9"></a>               Annotation = pData(target_demoData)[, ann_of_interest],</span>
<span id="cb5-10"><a href="#cb5-10"></a>               Q3 = unlist(apply(exprs(target_demoData), 2,</span>
<span id="cb5-11"><a href="#cb5-11"></a>                                 quantile, 0.75, na.rm = TRUE)),</span>
<span id="cb5-12"><a href="#cb5-12"></a>               NegProbe = exprs(target_demoData)[neg_probes, ])</span>
<span id="cb5-13"><a href="#cb5-13"></a>Stat_data_m &lt;- melt(Stat_data, measure.vars = c("Q3", "NegProbe"),</span>
<span id="cb5-14"><a href="#cb5-14"></a>                    variable.name = "Statistic", value.name = "Value")</span>
<span id="cb5-15"><a href="#cb5-15"></a></span>
<span id="cb5-16"><a href="#cb5-16"></a>plt1 &lt;- ggplot(Stat_data_m,</span>
<span id="cb5-17"><a href="#cb5-17"></a>               aes(x = Value, fill = Statistic)) +</span>
<span id="cb5-18"><a href="#cb5-18"></a>    geom_histogram(bins = 40) + theme_bw() +</span>
<span id="cb5-19"><a href="#cb5-19"></a>    scale_x_continuous(trans = "log2") +</span>
<span id="cb5-20"><a href="#cb5-20"></a>    facet_wrap(~Annotation, nrow = 1) + </span>
<span id="cb5-21"><a href="#cb5-21"></a>    scale_fill_brewer(palette = 3, type = "qual") +</span>
<span id="cb5-22"><a href="#cb5-22"></a>    labs(x = "Counts", y = "Segments, #")</span>
<span id="cb5-23"><a href="#cb5-23"></a></span>
<span id="cb5-24"><a href="#cb5-24"></a>plt2 &lt;- ggplot(Stat_data,</span>
<span id="cb5-25"><a href="#cb5-25"></a>               aes(x = NegProbe, y = Q3, color = Annotation)) +</span>
<span id="cb5-26"><a href="#cb5-26"></a>    geom_abline(intercept = 0, slope = 1, lty = "dashed", color = "darkgray") +</span>
<span id="cb5-27"><a href="#cb5-27"></a>    geom_point() + guides(color = "none") + theme_bw() +</span>
<span id="cb5-28"><a href="#cb5-28"></a>    scale_x_continuous(trans = "log2") + </span>
<span id="cb5-29"><a href="#cb5-29"></a>    scale_y_continuous(trans = "log2") +</span>
<span id="cb5-30"><a href="#cb5-30"></a>    theme(aspect.ratio = 1) +</span>
<span id="cb5-31"><a href="#cb5-31"></a>    labs(x = "Negative Probe GeoMean, Counts", y = "Q3 Value, Counts")</span>
<span id="cb5-32"><a href="#cb5-32"></a></span>
<span id="cb5-33"><a href="#cb5-33"></a>plt3 &lt;- ggplot(Stat_data,</span>
<span id="cb5-34"><a href="#cb5-34"></a>               aes(x = NegProbe, y = Q3 / NegProbe, color = Annotation)) +</span>
<span id="cb5-35"><a href="#cb5-35"></a>    geom_hline(yintercept = 1, lty = "dashed", color = "darkgray") +</span>
<span id="cb5-36"><a href="#cb5-36"></a>    geom_point() + theme_bw() +</span>
<span id="cb5-37"><a href="#cb5-37"></a>    scale_x_continuous(trans = "log2") + </span>
<span id="cb5-38"><a href="#cb5-38"></a>    scale_y_continuous(trans = "log2") +</span>
<span id="cb5-39"><a href="#cb5-39"></a>    theme(aspect.ratio = 1) +</span>
<span id="cb5-40"><a href="#cb5-40"></a>    labs(x = "Negative Probe GeoMean, Counts", y = "Q3/NegProbe Value, Counts")</span>
<span id="cb5-41"><a href="#cb5-41"></a></span>
<span id="cb5-42"><a href="#cb5-42"></a>btm_row &lt;- plot_grid(plt2, plt3, nrow = 1, labels = c("B", ""),</span>
<span id="cb5-43"><a href="#cb5-43"></a>                     rel_widths = c(0.43,0.57))</span>
<span id="cb5-44"><a href="#cb5-44"></a>plot_grid(plt1, btm_row, ncol = 1, labels = c("A", ""))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="normalization-1" class="slide level2">
<h2>Normalization <img data-src="figures/unnamed-chunk-32-1.png"></h2>
</section>
<section id="normalization-2" class="slide level2">
<h2>Normalization</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb6-1"><a href="#cb6-1"></a># visualize the first 10 segments with each normalization method</span>
<span id="cb6-2"><a href="#cb6-2"></a>boxplot(exprs(target_demoData)[,1:10],</span>
<span id="cb6-3"><a href="#cb6-3"></a>        col = "#9EDAE5", main = "Raw Counts",</span>
<span id="cb6-4"><a href="#cb6-4"></a>        log = "y", names = 1:10, xlab = "Segment",</span>
<span id="cb6-5"><a href="#cb6-5"></a>        ylab = "Counts, Raw")</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<img data-src="figures/unnamed-chunk-34-1.png" class="r-stretch"></section>
<section id="normalization-3" class="slide level2">
<h2>Normalization</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb7-1"><a href="#cb7-1"></a>boxplot(assayDataElement(target_demoData[,1:10], elt = "q_norm"),</span>
<span id="cb7-2"><a href="#cb7-2"></a>        col = "#2CA02C", main = "Q3 Norm Counts",</span>
<span id="cb7-3"><a href="#cb7-3"></a>        log = "y", names = 1:10, xlab = "Segment",</span>
<span id="cb7-4"><a href="#cb7-4"></a>        ylab = "Counts, Q3 Normalized")</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<img data-src="figures/unnamed-chunk-35-1.png" class="r-stretch"></section>
<section id="normalization-4" class="slide level2">
<h2>Normalization</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb8-1"><a href="#cb8-1"></a>boxplot(assayDataElement(target_demoData[,1:10], elt = "neg_norm"),</span>
<span id="cb8-2"><a href="#cb8-2"></a>        col = "#FF7F0E", main = "Neg Norm Counts",</span>
<span id="cb8-3"><a href="#cb8-3"></a>        log = "y", names = 1:10, xlab = "Segment",</span>
<span id="cb8-4"><a href="#cb8-4"></a>        ylab = "Counts, Neg. Normalized")</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<img data-src="figures/unnamed-chunk-36-1.png" class="r-stretch"></section>
<section id="unsupervised-analysis-and-umap-t-sne-plots" class="slide level2">
<h2>Unsupervised Analysis and UMAP &amp; t-SNE Plots</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb9-1"><a href="#cb9-1"></a>library(umap)</span>
<span id="cb9-2"><a href="#cb9-2"></a>library(Rtsne)</span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a># update defaults for umap to contain a stable random_state (seed)</span>
<span id="cb9-5"><a href="#cb9-5"></a>custom_umap &lt;- umap::umap.defaults</span>
<span id="cb9-6"><a href="#cb9-6"></a>custom_umap$random_state &lt;- 42</span>
<span id="cb9-7"><a href="#cb9-7"></a># run UMAP</span>
<span id="cb9-8"><a href="#cb9-8"></a>umap_out &lt;-</span>
<span id="cb9-9"><a href="#cb9-9"></a>    umap(t(log2(assayDataElement(target_demoData , elt = "q_norm"))),  </span>
<span id="cb9-10"><a href="#cb9-10"></a>         config = custom_umap)</span>
<span id="cb9-11"><a href="#cb9-11"></a>pData(target_demoData)[, c("UMAP1", "UMAP2")] &lt;- umap_out$layout[, c(1,2)]</span>
<span id="cb9-12"><a href="#cb9-12"></a>ggplot(pData(target_demoData),</span>
<span id="cb9-13"><a href="#cb9-13"></a>       aes(x = UMAP1, y = UMAP2, color = region, shape = class)) +</span>
<span id="cb9-14"><a href="#cb9-14"></a>    geom_point(size = 3) +</span>
<span id="cb9-15"><a href="#cb9-15"></a>    theme_bw()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="unsupervised-analysis-and-umap-t-sne-plots-1" class="slide level2">
<h2>Unsupervised Analysis and UMAP &amp; t-SNE Plots</h2>

<img data-src="figures/unnamed-chunk-37-1.png" class="r-stretch"></section>
<section id="unsupervised-analysis-and-umap-t-sne-plots-2" class="slide level2">
<h2>Unsupervised Analysis and UMAP &amp; t-SNE Plots</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb10-1"><a href="#cb10-1"></a># run tSNE</span>
<span id="cb10-2"><a href="#cb10-2"></a>set.seed(42) # set the seed for tSNE as well</span>
<span id="cb10-3"><a href="#cb10-3"></a>tsne_out &lt;-</span>
<span id="cb10-4"><a href="#cb10-4"></a>    Rtsne(t(log2(assayDataElement(target_demoData , elt = "q_norm"))),</span>
<span id="cb10-5"><a href="#cb10-5"></a>          perplexity = ncol(target_demoData)*.15)</span>
<span id="cb10-6"><a href="#cb10-6"></a>pData(target_demoData)[, c("tSNE1", "tSNE2")] &lt;- tsne_out$Y[, c(1,2)]</span>
<span id="cb10-7"><a href="#cb10-7"></a>ggplot(pData(target_demoData),</span>
<span id="cb10-8"><a href="#cb10-8"></a>       aes(x = tSNE1, y = tSNE2, color = region, shape = class)) +</span>
<span id="cb10-9"><a href="#cb10-9"></a>    geom_point(size = 3) +</span>
<span id="cb10-10"><a href="#cb10-10"></a>    theme_bw()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="unsupervised-analysis-and-umap-t-sne-plots-3" class="slide level2">
<h2>Unsupervised Analysis and UMAP &amp; t-SNE Plots</h2>

<img data-src="figures/unnamed-chunk-38-1.png" class="r-stretch"></section>
<section id="clustering-high-cv-genes-with-heatmap" class="slide level2">
<h2>Clustering High CV Genes with Heatmap</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb11-1"><a href="#cb11-1"></a>library(pheatmap)  # for pheatmap</span>
<span id="cb11-2"><a href="#cb11-2"></a># create a log2 transform of the data for analysis</span>
<span id="cb11-3"><a href="#cb11-3"></a>assayDataElement(object = target_demoData, elt = "log_q") &lt;-</span>
<span id="cb11-4"><a href="#cb11-4"></a>    assayDataApply(target_demoData, 2, FUN = log, base = 2, elt = "q_norm")</span>
<span id="cb11-5"><a href="#cb11-5"></a></span>
<span id="cb11-6"><a href="#cb11-6"></a># create CV function</span>
<span id="cb11-7"><a href="#cb11-7"></a>calc_CV &lt;- function(x) {sd(x) / mean(x)}</span>
<span id="cb11-8"><a href="#cb11-8"></a>CV_dat &lt;- assayDataApply(target_demoData,</span>
<span id="cb11-9"><a href="#cb11-9"></a>                         elt = "log_q", MARGIN = 1, calc_CV)</span>
<span id="cb11-10"><a href="#cb11-10"></a># show the highest CD genes and their CV values</span>
<span id="cb11-11"><a href="#cb11-11"></a>sort(CV_dat, decreasing = TRUE)[1:5]</span>
<span id="cb11-12"><a href="#cb11-12"></a>#&gt;   CAMK2N1    AKR1C1      AQP2     GDF15       REN </span>
<span id="cb11-13"><a href="#cb11-13"></a>#&gt; 0.5886006 0.5114973 0.4607206 0.4196469 0.4193216</span>
<span id="cb11-14"><a href="#cb11-14"></a></span>
<span id="cb11-15"><a href="#cb11-15"></a># Identify genes in the top 3rd of the CV values</span>
<span id="cb11-16"><a href="#cb11-16"></a>GOI &lt;- names(CV_dat)[CV_dat &gt; quantile(CV_dat, 0.8)]</span>
<span id="cb11-17"><a href="#cb11-17"></a>pheatmap(assayDataElement(target_demoData[GOI, ], elt = "log_q"),</span>
<span id="cb11-18"><a href="#cb11-18"></a>         scale = "row", </span>
<span id="cb11-19"><a href="#cb11-19"></a>         show_rownames = FALSE, show_colnames = FALSE,</span>
<span id="cb11-20"><a href="#cb11-20"></a>         border_color = NA,</span>
<span id="cb11-21"><a href="#cb11-21"></a>         clustering_method = "average",</span>
<span id="cb11-22"><a href="#cb11-22"></a>         clustering_distance_rows = "correlation",</span>
<span id="cb11-23"><a href="#cb11-23"></a>         clustering_distance_cols = "correlation",</span>
<span id="cb11-24"><a href="#cb11-24"></a>         breaks = seq(-3, 3, 0.05),</span>
<span id="cb11-25"><a href="#cb11-25"></a>         color = colorRampPalette(c("purple3", "black", "yellow2"))(120),</span>
<span id="cb11-26"><a href="#cb11-26"></a>         annotation_col = </span>
<span id="cb11-27"><a href="#cb11-27"></a>             pData(target_demoData)[, c("class", "segment", "region")])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="clustering-high-cv-genes-with-heatmap-1" class="slide level2">
<h2>Clustering High CV Genes with Heatmap</h2>

<img data-src="figures/unnamed-chunk-39-1.png" class="r-stretch"></section>
<section id="visualizing-de-genes-with-volcano-plot" class="slide level2">
<h2>Visualizing DE Genes with Volcano Plot</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb12-1"><a href="#cb12-1"></a>library(ggrepel) </span>
<span id="cb12-2"><a href="#cb12-2"></a># Categorize Results based on P-value &amp; FDR for plotting</span>
<span id="cb12-3"><a href="#cb12-3"></a>results$Color &lt;- "NS or FC &lt; 0.5"</span>
<span id="cb12-4"><a href="#cb12-4"></a>results$Color[results$`Pr(&gt;|t|)` &lt; 0.05] &lt;- "P &lt; 0.05"</span>
<span id="cb12-5"><a href="#cb12-5"></a>results$Color[results$FDR &lt; 0.05] &lt;- "FDR &lt; 0.05"</span>
<span id="cb12-6"><a href="#cb12-6"></a>results$Color[results$FDR &lt; 0.001] &lt;- "FDR &lt; 0.001"</span>
<span id="cb12-7"><a href="#cb12-7"></a>results$Color[abs(results$Estimate) &lt; 0.5] &lt;- "NS or FC &lt; 0.5"</span>
<span id="cb12-8"><a href="#cb12-8"></a>results$Color &lt;- factor(results$Color,</span>
<span id="cb12-9"><a href="#cb12-9"></a>                        levels = c("NS or FC &lt; 0.5", "P &lt; 0.05",</span>
<span id="cb12-10"><a href="#cb12-10"></a>                                   "FDR &lt; 0.05", "FDR &lt; 0.001"))</span>
<span id="cb12-11"><a href="#cb12-11"></a></span>
<span id="cb12-12"><a href="#cb12-12"></a># pick top genes for either side of volcano to label</span>
<span id="cb12-13"><a href="#cb12-13"></a># order genes for convenience:</span>
<span id="cb12-14"><a href="#cb12-14"></a>results$invert_P &lt;- (-log10(results$`Pr(&gt;|t|)`)) * sign(results$Estimate)</span>
<span id="cb12-15"><a href="#cb12-15"></a>top_g &lt;- c()</span>
<span id="cb12-16"><a href="#cb12-16"></a>for(cond in c("DKD", "normal")) {</span>
<span id="cb12-17"><a href="#cb12-17"></a>    ind &lt;- results$Subset == cond</span>
<span id="cb12-18"><a href="#cb12-18"></a>    top_g &lt;- c(top_g,</span>
<span id="cb12-19"><a href="#cb12-19"></a>               results[ind, 'Gene'][</span>
<span id="cb12-20"><a href="#cb12-20"></a>                   order(results[ind, 'invert_P'], decreasing = TRUE)[1:15]],</span>
<span id="cb12-21"><a href="#cb12-21"></a>               results[ind, 'Gene'][</span>
<span id="cb12-22"><a href="#cb12-22"></a>                   order(results[ind, 'invert_P'], decreasing = FALSE)[1:15]])</span>
<span id="cb12-23"><a href="#cb12-23"></a>}</span>
<span id="cb12-24"><a href="#cb12-24"></a>top_g &lt;- unique(top_g)</span>
<span id="cb12-25"><a href="#cb12-25"></a>results &lt;- results[, -1*ncol(results)] # remove invert_P from matrix</span>
<span id="cb12-26"><a href="#cb12-26"></a></span>
<span id="cb12-27"><a href="#cb12-27"></a># Graph results</span>
<span id="cb12-28"><a href="#cb12-28"></a>ggplot(results,</span>
<span id="cb12-29"><a href="#cb12-29"></a>       aes(x = Estimate, y = -log10(`Pr(&gt;|t|)`),</span>
<span id="cb12-30"><a href="#cb12-30"></a>           color = Color, label = Gene)) +</span>
<span id="cb12-31"><a href="#cb12-31"></a>    geom_vline(xintercept = c(0.5, -0.5), lty = "dashed") +</span>
<span id="cb12-32"><a href="#cb12-32"></a>    geom_hline(yintercept = -log10(0.05), lty = "dashed") +</span>
<span id="cb12-33"><a href="#cb12-33"></a>    geom_point() +</span>
<span id="cb12-34"><a href="#cb12-34"></a>    labs(x = "Enriched in Tubules &lt;- log2(FC) -&gt; Enriched in Glomeruli",</span>
<span id="cb12-35"><a href="#cb12-35"></a>         y = "Significance, -log10(P)",</span>
<span id="cb12-36"><a href="#cb12-36"></a>         color = "Significance") +</span>
<span id="cb12-37"><a href="#cb12-37"></a>    scale_color_manual(values = c(`FDR &lt; 0.001` = "dodgerblue",</span>
<span id="cb12-38"><a href="#cb12-38"></a>                                  `FDR &lt; 0.05` = "lightblue",</span>
<span id="cb12-39"><a href="#cb12-39"></a>                                  `P &lt; 0.05` = "orange2",</span>
<span id="cb12-40"><a href="#cb12-40"></a>                                  `NS or FC &lt; 0.5` = "gray"),</span>
<span id="cb12-41"><a href="#cb12-41"></a>                       guide = guide_legend(override.aes = list(size = 4))) +</span>
<span id="cb12-42"><a href="#cb12-42"></a>    scale_y_continuous(expand = expansion(mult = c(0,0.05))) +</span>
<span id="cb12-43"><a href="#cb12-43"></a>    geom_text_repel(data = subset(results, Gene %in% top_g &amp; FDR &lt; 0.001),</span>
<span id="cb12-44"><a href="#cb12-44"></a>                    size = 4, point.padding = 0.15, color = "black",</span>
<span id="cb12-45"><a href="#cb12-45"></a>                    min.segment.length = .1, box.padding = .2, lwd = 2,</span>
<span id="cb12-46"><a href="#cb12-46"></a>                    max.overlaps = 50) +</span>
<span id="cb12-47"><a href="#cb12-47"></a>    theme_bw(base_size = 16) +</span>
<span id="cb12-48"><a href="#cb12-48"></a>    theme(legend.position = "bottom") +</span>
<span id="cb12-49"><a href="#cb12-49"></a>    facet_wrap(~Subset, scales = "free_y")</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="visualizing-de-genes-with-volcano-plot-1" class="slide level2">
<h2>Visualizing DE Genes with Volcano Plot</h2>

<img data-src="figures/unnamed-chunk-44-1.png" class="r-stretch"></section>
<section id="visualizing-de-genes-with-ma-plot" class="slide level2">
<h2>Visualizing DE Genes with MA Plot</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb13-1"><a href="#cb13-1"></a>results$MeanExp &lt;-</span>
<span id="cb13-2"><a href="#cb13-2"></a>    rowMeans(assayDataElement(target_demoData,</span>
<span id="cb13-3"><a href="#cb13-3"></a>                              elt = "q_norm"))</span>
<span id="cb13-4"><a href="#cb13-4"></a></span>
<span id="cb13-5"><a href="#cb13-5"></a>top_g2 &lt;- results$Gene[results$Gene %in% top_g &amp;</span>
<span id="cb13-6"><a href="#cb13-6"></a>                           results$FDR &lt; 0.001 &amp;</span>
<span id="cb13-7"><a href="#cb13-7"></a>                           abs(results$Estimate) &gt; .5 &amp;</span>
<span id="cb13-8"><a href="#cb13-8"></a>                           results$MeanExp &gt; quantile(results$MeanExp, 0.9)]</span>
<span id="cb13-9"><a href="#cb13-9"></a></span>
<span id="cb13-10"><a href="#cb13-10"></a>ggplot(subset(results, !Gene %in% neg_probes),</span>
<span id="cb13-11"><a href="#cb13-11"></a>       aes(x = MeanExp, y = Estimate,</span>
<span id="cb13-12"><a href="#cb13-12"></a>           size = -log10(`Pr(&gt;|t|)`),</span>
<span id="cb13-13"><a href="#cb13-13"></a>           color = Color, label = Gene)) +</span>
<span id="cb13-14"><a href="#cb13-14"></a>    geom_hline(yintercept = c(0.5, -0.5), lty = "dashed") +</span>
<span id="cb13-15"><a href="#cb13-15"></a>    scale_x_continuous(trans = "log2") +</span>
<span id="cb13-16"><a href="#cb13-16"></a>    geom_point(alpha = 0.5) + </span>
<span id="cb13-17"><a href="#cb13-17"></a>    labs(y = "Enriched in Glomeruli &lt;- log2(FC) -&gt; Enriched in Tubules",</span>
<span id="cb13-18"><a href="#cb13-18"></a>         x = "Mean Expression",</span>
<span id="cb13-19"><a href="#cb13-19"></a>         color = "Significance") +</span>
<span id="cb13-20"><a href="#cb13-20"></a>    scale_color_manual(values = c(`FDR &lt; 0.001` = "dodgerblue",</span>
<span id="cb13-21"><a href="#cb13-21"></a>                                  `FDR &lt; 0.05` = "lightblue",</span>
<span id="cb13-22"><a href="#cb13-22"></a>                                  `P &lt; 0.05` = "orange2",</span>
<span id="cb13-23"><a href="#cb13-23"></a>                                  `NS or FC &lt; 0.5` = "gray")) +</span>
<span id="cb13-24"><a href="#cb13-24"></a>    geom_text_repel(data = subset(results, Gene %in% top_g2),</span>
<span id="cb13-25"><a href="#cb13-25"></a>                    size = 4, point.padding = 0.15, color = "black",</span>
<span id="cb13-26"><a href="#cb13-26"></a>                    min.segment.length = .1, box.padding = .2, lwd = 2) +</span>
<span id="cb13-27"><a href="#cb13-27"></a>    theme_bw(base_size = 16) +</span>
<span id="cb13-28"><a href="#cb13-28"></a>    facet_wrap(~Subset, nrow = 2, ncol = 1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="visualizing-de-genes-with-ma-plot-1" class="slide level2">
<h2>Visualizing DE Genes with MA Plot</h2>

<img data-src="figures/unnamed-chunk-45-1.png" class="r-stretch"></section>
<section id="visualizing-de-genes-with-heatmap" class="slide level2">
<h2>Visualizing DE Genes with Heatmap</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode numberSource {{r}} number-lines code-with-copy"><code class="sourceCode"><span id="cb14-1"><a href="#cb14-1"></a># select top significant genes based on significance, plot with pheatmap</span>
<span id="cb14-2"><a href="#cb14-2"></a>GOI &lt;- unique(subset(results, `FDR` &lt; 0.001)$Gene)</span>
<span id="cb14-3"><a href="#cb14-3"></a>pheatmap(log2(assayDataElement(target_demoData[GOI, ], elt = "q_norm")),</span>
<span id="cb14-4"><a href="#cb14-4"></a>         scale = "row", </span>
<span id="cb14-5"><a href="#cb14-5"></a>         show_rownames = FALSE, show_colnames = FALSE,</span>
<span id="cb14-6"><a href="#cb14-6"></a>         border_color = NA,</span>
<span id="cb14-7"><a href="#cb14-7"></a>         clustering_method = "average",</span>
<span id="cb14-8"><a href="#cb14-8"></a>         clustering_distance_rows = "correlation",</span>
<span id="cb14-9"><a href="#cb14-9"></a>         clustering_distance_cols = "correlation",</span>
<span id="cb14-10"><a href="#cb14-10"></a>         cutree_cols = 2, cutree_rows = 2,</span>
<span id="cb14-11"><a href="#cb14-11"></a>         breaks = seq(-3, 3, 0.05),</span>
<span id="cb14-12"><a href="#cb14-12"></a>         color = colorRampPalette(c("purple3", "black", "yellow2"))(120),</span>
<span id="cb14-13"><a href="#cb14-13"></a>         annotation_col = pData(target_demoData)[, c("region", "class")])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="visualizing-de-genes-with-heatmap-1" class="slide level2">
<h2>Visualizing DE Genes with Heatmap</h2>

<img data-src="figures/unnamed-chunk-47-1.png" class="r-stretch"></section>
<section id="conclusion" class="slide level2">
<h2>Conclusion</h2>
</section>
<section id="learned-insteresting-skills" class="slide level2">
<h2>Learned Insteresting Skills</h2>
<ul>
<li>How to plot UMAP &amp; t-SNE Plots</li>
<li>How to plot volcano plot and MA plot</li>
<li>How to plot heatmap using pheat packages</li>
<li>QC &amp; processing and normalization</li>
</ul>
</section>
<section id="challeges" class="slide level2">
<h2>Challeges</h2>
<ul>
<li>Install the Packages (Enviorment)</li>
<li>Load the Data</li>
<li>Understand the object and related functions from the packages</li>
<li>QC &amp; preprocessing</li>
</ul>
<div class="footer footer-default">

</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="geomx_presentation_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="geomx_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="geomx_presentation_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="geomx_presentation_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        setTimeout(function() {
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const cites = ref.parentNode.getAttribute('data-cites').split(' ');
        tippyHover(ref, function() {
          var popup = window.document.createElement('div');
          cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    });
    </script>
    

</body></html>