---
title: "Analyzing GeoMx-NGS RNA Expression Data with GeomxTools"
author: Jason Reeves1, Prajan Divakar1, Nicole Ortogero1, Maddy Griswold1, Zhi Yang1, Stephanie Zimmerman1, Rona Vitancol1 and David Henderson
format: 
  revealjs:
   theme: moon
editor: visual
---

# **Introduction**

What is spatial profiling and why perform spatial profiling?

::: notes
Bulk profiling approaches such as RNA-seq lose information about the relative composition of the tissue. Even techniques which excel at identifying rare cell populations such as single cell RNA-seq, lose information about where in the tissue those cells are located. Spatial profiling enables deep phenotpyic profiling and mapping of those profiles to specific locations within the tissue context. This is especially powerful when the processes under study are unevenly distributed in the tissue (like patchy tissue disease).
:::

# NanoString GeoMx DSP

![](GEOMx-NGS_files/figure-revealjs/GeoMx-DSP-workflow-Cellecta-r1-2000x.png){width="445"}

::: notes
After hybridization of specially oligo-conjugated probes to a section, the tissue is counterstained and imaged using immunofluorescence microscopy. Regions of interest are selected. An ultraviolet laser pulse is focused precisely on the ROI to release oligos conjugated to the target protein or RNAs. The DSP intrument aspirates these oligos into collection wells. The oligos contain barcodes which map them to their source protein/RNA target, digital counting of these oligos with nCounter or next generation sequencing permits a direct readout of target abundance localized to the specific ROI with the tissue section.
:::

## **Motivation & Scope**

The motivation for this vignette is to enable scientists to work with GeoMx-NGS gene expression data and understand a standard data analysis workflow.

::: notes
Speaker notes go here
:::

## **Data**

The dataset includes 4 diabetic kidney disease (DKD) and 3 healthy kidney tissue samples. Regions of interest (ROI) were spatially profiled to focus on two different kidney structures: tubules or glomeruli. One glomerular ROI contains the entirety of a single glomerulus. Each tubular ROI contains multiple tubules that were segmented into distal (PanCK+) and proximal (PanCK-) tubule areas of illumination (AOI).

::: notes
Speaker notes go here
:::

# QC and Pre-processing

## **Segment QC**

::: r-stack
![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-8-1.png){.fragment width="1950" height="550"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-9-1.png){.fragment width="1950" height="550"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-10-1.png){.fragment width="1950" height="550"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-12-1.png){.fragment width="1950" height="550"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-13-1.png){.fragment width="1950" height="550"}
:::

## **Probe QC**

A probe is removed globally from the dataset if either of the following is true:

-   the geometric mean of that probe's counts from all segments divided by the geometric mean of all probe counts representing the target from all segments is less than 0.1

-   the probe is an outlier according to the Grubb's test in at least 20% of the segments

::: notes
Before the data was summarized into gene-level count data, low-performing probes were removed. Basically this QC is an outlier removal process, whereby probes are either removed entirely from the study or from specific segments.
:::

## Limit of Quantification

LOQi=geomean(NegProbei)∗geoSD(NegProbei)n

::: notes
The LOQ is calculated based on the distribution of negative control probes ans is intended to approximate the quantifiable limit of gene expression per segment.
:::

## **Segment Gene Detection**

We first filter out segments with exceptionally low signal.

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-25-1.png)

::: notes
Speaker notes go here
:::

## **Gene Filtering**

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-30-1.png)

::: notes
Here they graphed the total number of genes detected in different percentages of segments. Using this visualization, they can better understand global gene detection in their study and select how many low detected genes to filter out of the dataset. Gene filtering increases performances of downstream statistical tests and improves interpretation of true biological signal. From this they decided to focus on the genes detected in at least 10% of the segments, they filtered out the remainder of the targets.
:::

## **Normalization**

We will now normalize the GeoMx data for downstream visualizations and differential expression.

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-32-1.png)

::: notes
Here the researchers explored the relationship between the upper quartile (Q3) of the counts in each segment with the geometric mean of the negative control probes in the data.
:::

## Effects of Normalization

We will use Q3 normalized data moving forward. We use the `normalize` function from `NanoStringNCTools` to create normalization factors reflecting each data type. 

::: r-stack
![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-34-1.png){.fragment width="1150" height="450"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-35-1.png){.fragment width="1150" height="450"}

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-36-1.png){.fragment width="1150" height="450"}
:::

## **Visualizing DE Genes**

A canonical visualization for interpreting differential gene expression results is the volcano plot. 

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-44-1.png)

::: notes
Speaker notes go here
:::

## **Plotting Genes of Interest**

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-38-1.png)

::: notes
Speaker notes go here
:::

## **Heatmap of Significant Genes**

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-48-1.png)

::: notes
Speaker notes go here
:::

## **Additional Analysis Methods**

### **MA Plot**

![](GEOMx-NGS_files/figure-revealjs/unnamed-chunk-49-1.png)

::: notes
Speaker notes go here
:::

## Interesting thoughts

::: notes
Speaker notes go here
:::

## Challenges

::: notes
Speaker notes go here
:::
